### DISCUSSION TITLE
FritzOS 7.9 inhaus

### DISCUSSION THREAD

**Discussion by fda77** [Freetz-NG/freetz-ng#961] [Allgemein]
Ein paar Gedanken zum FOS 8

 - es gibt kein CONFIG_NTFS mehr in der rc.conf, womit realisiert avm ntfs mittlerweile? Oder ist das ganz rausgefolgen?
 - tr069fwupdate binary gibts es nicht mehr (umbenannt nach fwupdate?)
 - avm nutzt busybox 1.35 (6670) und 1.36.1 (7.90)
 - httpsdl binary ist auch weg (umbenannt nach webget?, mehr/weniger features?)
 - das squashfs der 7590 ac+ax ist nun little-endian - booten die freetz-images noch?
 - es gibt bestimmt noch viele andere Änderungen, die jemand finden müsste

**Comment by McBane87** [comment]
> * das squashfs der 7590 ac+ax ist nun little-endian - booten die freetz-images noch?

Meine 7590 (nicht AX), bootet mit dem 7.9er Image (`7590_07.90.all-rev111682_inhaus_freetz-ng-23449M-943a6f967_20240402-124514.image`) und bleibt auch aktiv. Also ohne Loops. Was mir spontan aufgefallen ist:

Beim `make` musste ich das `patches/scripts/191-add_responsive_webmenu.sh` deaktivieren, weil es einen Fehler geworfen hat 
  ```
applying patch file ./patches/scripts/191-add_responsive_webmenu.sh
  adding responsive webmenu link
    patching build/modified/filesystem/usr/lua/menu_data.lua
ERROR: modsed failed editing build/modified/filesystem/usr/lua/menu_data.lua
```

Im FOS Webinterface komme ich nicht weiter. Lande direkt bei:
```
Willkommen bei Ihrer FRITZ!Box

Unsere Datenschutzerklärung

Diagnose und Wartung
Zur Verbesserung Ihres Produktes und für den sicheren Betrieb an Ihrem Anschluss verwendet AVM Fehlerberichte und bei Bedarf
Diagnosedaten. So profitieren Sie von Optimierungen bei Updates oder notwendigen Anpassungen durch AVM. In Laborversionen sind
diese Dienste daher immer aktiviert. In Release-Versionen können Sie hier Anpassungen vornehmen.

                                                                                                                      [Weiter]
```

Wenn ich dann weiter klicke, lädt es kurz die `index.lua` und wirft mich dann sofort wieder zurück zu obiger Diagnose und Wartung Abfrage. Ist also eine endlose Schleife. Vielleicht liegts daran, dass ich vorher kein Werksreset gemacht habe, aber da es sich um meine Spielebox handelt, wo fast nix konfiguriert ist, hab ich das halt weggelassen.

**Reply by fda77** [reply]
Danke fürs testen! Gute zu wissen dass das Image überhaupt noch bootet
 - Bislang war die Endianness des Squashfs bei AVM immer die gleiche wie das Linux selbst. Ab dieser Inhaus muss das getrennt ermittelt werden: 98ad0bc0adfcb944f080c3cc3630edb63f873d63 und ganz am Ende von "make" sieht man auch nach "creating filesystem image" wie es gepackt wurde: dd9ebda4b4e82447eede6c3ebc2a16c975ebb684
 - Hast du versucht Cache, Cookies etc beim Browser zu leeren? Oder einfach einen anderen genommen? Das FOS hatte in die Richtung schon öfter mal Probleme gemacht

**Reply by fda77** [reply]
- ab7eea0f6

**Reply by McBane87** [reply]
> und ganz am Ende von "make" sieht man auch nach "creating filesystem image" wie es gepackt wurde

Ja, da steht dann, bei mir: `creating filesystem image (SquashFS4_le-xz)`

> Hast du versucht Cache, Cookies etc beim Browser zu leeren? Oder einfach einen anderen genommen?

Jetzt ja, aber keine Änderung. Sowohl im Inkognito Modus, als auch mit anderem Browser das gleiche Fehlerbild.

> https://github.com/Freetz-NG/freetz-ng/commit/ab7eea0f682194954fa74bb9bc003aad82ea140b

Danke. Mit der Anpassung funktioniert der Patch wieder.

**Reply by fda77** [reply]
> Im FOS Webinterface komme ich nicht weiter

Keine Ahnung. Vielleich ists ein avm fehler. Das findet man heraus wenn man ohne freetz flasht. Ansonsten könnte der webserver crashen, das sieht man meist im syslog.
Oder irgend ein anderer freetz-patch passt nicht.

Da meine neueste Fritzbox FOS 7.5x hat kann ich zu 7.8 oder gar zu 7.9 überhaupt nichts sagen. Und mit dieser Fritzbox teste ich auch nicht
